AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m

```diff
@@ -81,17 +81,11 @@ @implementation MS_Reachability {
 #pragma clang diagnostic ignored "-Wnullable-to-nonnull-conversion"
 
 + (instancetype)reachabilityWithHostName:(NSString *)hostName {
-  __block MS_Reachability *returnValue = NULL;
+  MS_Reachability *returnValue = NULL;
   SCNetworkReachabilityRef reachability =
       SCNetworkReachabilityCreateWithName(NULL, [hostName UTF8String]);
   if (reachability != NULL) {
-    if ([NSThread isMainThread]) {
-      returnValue = [[MS_Reachability alloc] init];
-    } else {
-      dispatch_sync(dispatch_get_main_queue(), ^{
-        returnValue = [[MS_Reachability alloc] init];
-      });
-    }
+    returnValue = [[MS_Reachability alloc] init];
```

Comment:
Yep I'd love do delete this code too 👍 . Though we might need some other sort of synchronization here, I have a vague memory of reachability not being thread safe (cc: @jaelim-ms).

Reply:
Yep, Reachability is not thread safe.

Reply:
Talked to @MatkovIvan and this is only for the init, all other reachability code is still on main thread. We should be good.

Reply:
I'll unblock it seems like it's not using any shared resources at this time.

Reply:
Do we know if `SCNetworkReachabilityCreateWithName(NULL, [hostName UTF8String])` returns a different reference every time? If not is it thread safe? @MatkovIvan

Reply:
According to the docs, it returns a new, immutable reference:
`A new immutable reachability reference. You must release the returned value.`

Reply:
Thanks @ElektrojungeMS. Anyway, this isn't changed in the PR.

Reply:
Yeah that looks good.